### ISSUE TITLE
Command to look up codepoint from a character

### ISSUE THREAD

**Issue by NotTheDr01ds** [nushell/nushell#14264]
### Related problem

Related: #14174 - The `seq char` command doesn't work well for non-alpha characters.

Rather than fixing `seq char`, what if we deprecate it (or move a new version to the Standard Library.

The problem at the moment is that (AFAICT) we have no way to *get* the codepoint for a character in order to know where it lies in the ASCII (or Unicode) table.

### Describe the solution you'd like

Either a new command or an extension to the `char` command to be able to look up the codepoint for a character.  For example:

```nu
let start = (char --int-from-char 'A')
$start
# => 65
let end = (char --int-from-char 'D')
$end
# => 68

# Replace the `seq char` with:
$start..$end | each {char --integer $in}
# => ╭───┬───╮
# => │ 0 │ A │
# => │ 1 │ B │
# => │ 2 │ C │
# => │ 3 │ D │
# => ╰───┴───╯
```

**Comment by NotTheDr01ds** [comment]
Side-note:  For ASCII (0..127) values, it's fairly easy to convert:

```nu
'A' | into binary | into int
# => 65
' ' | into binary | into int
# => 32
```

But characters 128 and greater don't work.

**Comment by fdncred** [comment]
Seems like some work on the `char` command may be in order. Maybe unifying the `-i` and `-u` and make it also work in this scenario. However, I'm not a fan of the idea of removing `seq char`.

**Comment by NotTheDr01ds** [comment]
As mentioned, it could be moved to the standard library.  Is it important enough to be a built-in?

**Comment by fdncred** [comment]
yup, i saw that. i'd personally rather have it as a rust command.

**Comment by NotTheDr01ds** [comment]
>Maybe if we "fix" char to accomodate it, I'd be more open to it.

Turns out there's not really a need for it after all.

```nu
use std/assert
alias "original seq char" = seq char

def "seq char" [start: string, end: string] {
  let start_code = $start | into binary | into int
  let end_code = $end | into binary | into int
  if not (($start_code in 32..126) and ($end_code in 32..126)) {
    error make {msg: 'Start and End values must be printable ASCII characters'}
  }

  ($start_code)..($end_code) | each { char -i $in }
}

# Behaves as the original version
seq char A C
# => ╭───┬───╮
# => │ 0 │ A │
# => │ 1 │ B │
# => │ 2 │ C │
# => ╰───┴───╯

assert equal (
  seq char A Z
) (
  original seq char A Z
)
# => passes

assert equal (
  seq char a z
) (
  original seq char a z
)
# => passes

# Also fixes the issues in #14174
seq char '0' '4'
# => ╭───┬───╮
# => │ 0 │ 0 │
# => │ 1 │ 1 │
# => │ 2 │ 2 │
# => │ 3 │ 3 │
# => │ 4 │ 4 │
# => ╰───┴───╯

seq char ' ' '~'
# => Table of all printable ASCII characters

# Even works in reverse
seq char 9 0
```

**Comment by fdncred** [comment]
nice work on the script, i like it! I'm trying to do '32' to '126' to test your range and it's failing. It may also be better to check them separately so that an error could be produced saying that the start of the range is wrong, or the end of the range.

**Comment by NotTheDr01ds** [comment]
@fdncred Huh - looks like an outdated version got over-pasted.  I had already separated them out into different errors.  I'll repaste, but that seemed to break formatting.

**Comment by NotTheDr01ds** [comment]
@fdncred Oh, did we lose `nu` as a language specifier?

**Comment by fdncred** [comment]
> @fdncred Oh, did we lose `nu` as a language specifier?

On github, it's always been nushell I believe but in the documentation I think we can use nu or nushell.

**Comment by NotTheDr01ds** [comment]
> I'm trying to do '32' to '126' to test your range and it's failing

Are you doing '32' to '126' expecting it to work?  That's the `seq char ' ' '~'`

**Comment by fdncred** [comment]
> > I'm trying to do '32' to '126' to test your range and it's failing
> 
> Are you doing '32' to '126' expecting it to work? That's the `seq char ' ' '~'`

Yup, I was. I can see how that's wrong now. Was just trying to test your old code. This new code is very different from the old.

**Comment by fdncred** [comment]
Also, I'm not sure this counts `seq char '0' '4'` when there are more than 5 characters. For instance, you can't do `seq char '0' '10'` but maybe we could tweak on that over time. Not sure why anyone would do that but if we advertise that we support it, we should.

**Comment by NotTheDr01ds** [comment]
> but if we advertise that we support it, we should.

I'm not sure that we do?  We only allow a single character (both in the original version and this script), right?

**Comment by NotTheDr01ds** [comment]
For `seq char '0' '4'`, I expect and get the *strings* 0, 1, 2, 3, and 4.  Is that what you are expecting and get?

**Comment by NotTheDr01ds** [comment]
> On github, it's always been nushell

Heh - I've been doing it wrong for a while then, I think.

**Comment by NotTheDr01ds** [comment]
> This new code is very different from the old.

Yeah, without the error handling, it's really only three lines.

**Comment by fdncred** [comment]
> For `seq char '0' '4'`, I expect and get the _strings_ 0, 1, 2, 3, and 4. Is that what you are expecting and get?

Yes, that's what I expected and got but what I'm saying is I wonder if there should be some error handling that is like, "Hey, you're trying to do numbers. Maybe you should try the seq command instead", when the range is like '0'..'10' or larger.

**Comment by anomius** [comment]
> Yes, that's what I expected and got but what I'm saying is I wonder if there should be some error handling that is like, "Hey, you're trying to do numbers. Maybe you should try the seq command instead", when the range is like '0'..'10' or larger.

Should the seq be type-agnostic? Or will it be a mess, like JavaScript?

**Comment by fdncred** [comment]
> > Yes, that's what I expected and got but what I'm saying is I wonder if there should be some error handling that is like, "Hey, you're trying to do numbers. Maybe you should try the seq command instead", when the range is like '0'..'10' or larger.
> 
> Should the seq be type-agnostic? Or will it be a mess, like JavaScript?

I don't think so. I like `seq` for numbers, `seq char` for characters and `seq date` for dates. Maybe it should be `seq num` for numbers so we can have a root command `seq` that lists all the seq commands?

**Comment by fdncred** [comment]
looks good!
